### PR TITLE
added-implementation-of-email-service-in-separate-branch

### LINKED ISSUE #109: Create email notification service

### PR DESCRIPTION
**Pull request by bogdangavrylyak**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: CarBackEnd/CarBackEnd/appsettings.Development.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "ConnectionStrings": {
-    "CarConnection": "Server=WINDOWS-HKJEEGV\\SQLEXPRESS;Database=Car;Trusted_Connection=True;MultipleActiveResultSets=true"
+    "CarConnection": "Server=(localdb)\\MSSQLLocalDB;Database=Car;Trusted_Connection=True;MultipleActiveResultSets=true"
```
Comment (BorysMalynovskyi) [inline CarBackEnd/CarBackEnd/appsettings.Development.json:3]: Infrastructure change. Must not be here

File: CarBackEnd/RazorClassLibraryForEmails/Views/Shared/EmailLayout.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace RazorClassLibraryForEmails.Views.Shared
+{
+    public class EmailLayoutModel : PageModel
+    {
+        public void OnGet()
```
Comment (BorysMalynovskyi) [inline CarBackEnd/RazorClassLibraryForEmails/Views/Shared/EmailLayout.cshtml.cs:12]: What is the method that does nothing?

File: CarBackEnd/RazorClassLibraryForEmails/Views/Emails/CancelJourney/CancelJourney.cshtml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+<br />
+
+@if (Model.Attachments != null)
+{
+    <img src=cid:imgpath height=100 width=100 alt="Image" />
+}
+
+
+<br />
+
```
Comment (BorysMalynovskyi) [inline CarBackEnd/RazorClassLibraryForEmails/Views/Emails/CancelJourney/CancelJourney.cshtml:22]: Please remove emplty lines

File: CarBackEnd/RazorClassLibraryForEmails/Views/Emails/CancelJourney/CancelJourney.cshtml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+<p>
```
Comment (BorysMalynovskyi) [inline CarBackEnd/RazorClassLibraryForEmails/Views/Emails/CancelJourney/CancelJourney.cshtml:23]: Not used p tag

File: CarBackEnd/RazorClassLibraryForEmails/Services/RazorViewToStringRenderer.cs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+        private ActionContext GetActionContext()
+        {
+            var httpContext = new DefaultHttpContext();
+            httpContext.RequestServices = _serviceProvider;
+            return new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
+        }
+
+    }
+    public interface IRazorViewToStringRenderer
```
Comment (BorysMalynovskyi) [inline CarBackEnd/RazorClassLibraryForEmails/Services/RazorViewToStringRenderer.cs:91]: Interfaces should be in separate files

**Review by BorysMalynovskyi — COMMENTED** [review]
Please go through the comments and update accordingly

File: CarBackEnd/Car.BLL/Car.BLL.csproj [hunk 6]
```diff
@@ -10,6 +10,7 @@
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/Car.BLL/Car.BLL.csproj:10]: I've seen a similar PR [here](https://github.com/ita-social-projects/Car-Back-End/pull/115), is it still actual?

**Pull request closed without merging**